src/plugin/kubearmor.go

```diff
@@ -581,101 +585,94 @@ func GetNetworkLogsFromKubeArmor() []*pb.Alert {
+	// Direction
+	if strings.Contains(kaNwLog.Data, "tcp_") {
+		knoxNetLog.Protocol = libs.IPProtocolTCP
+		if strings.Contains(kaNwLog.Data, "tcp_accept") {
+			knoxNetLog.Direction = "INGRESS"
+		} else if strings.Contains(kaNwLog.Data, "tcp_connect") {
+			knoxNetLog.Direction = "EGRESS"
+		}
 
-			if ip != destPod && (strings.Contains(destPod, "pod") || strings.Contains(destPod, "svc")) {
-				locKnoxLog.DstPodName = strings.Split(destPod, "/")[1]
-				locKnoxLog.DstReservedLabels = strings.Split(destLabels, ",")
-				locKnoxLog.DstNamespace = destNs
+		// Extract IP and port
+		resslice := strings.Split(kaNwLog.Resource, " ")
+		for _, locres := range resslice {
+			if strings.Contains(locres, "remoteip") {
+				ip = strings.Split(locres, "=")[1]
 			}
-			locKnoxLog.DstIP = ip
-			locKnoxLog.DstPort, _ = strconv.Atoi(port)
-			locKnoxLog.SynFlag = true
-		} else if strings.Contains(kalog.Data, "SYS_BIND") {
-			var port string
-			// TODO : Identify a way to get protocol from kubearmor
-			// locKnoxLog.Protocol = libs.IPProtocolUDP
-
-			resSlice := strings.Split(kalog.Resource, " ")
-			for _, v := range resSlice {
-				if strings.Contains(v, "sin_port") {
-					port = strings.Split(v, "=")[1]
-				}
+			if strings.Contains(locres, "port") {
+				port = strings.Split(locres, "=")[1]
 			}
-			//locKnoxLog.DstIP = "0.0.0.0"
-			locKnoxLog.DstPort, _ = strconv.Atoi(port)
-			locKnoxLog.Direction = "INGRESS"
-		} else if strings.Contains(kalog.Data, "SYS_SOCKET") && strings.Contains(kalog.Resource, "SOCK_DGRAM") {
-			locKnoxLog.Protocol = libs.IPProtocolUDP
-			//locKnoxLog.DstIP = "0.0.0.0"
-			locKnoxLog.Direction = "EGRESS"
 		}
 
-		if kalog.Result != "Passed" {
-			locKnoxLog.Action = "Deny"
-		} else {
-			locKnoxLog.Action = "Allow"
+		if net.ParseIP(ip).IsLoopback() {
+			// ignore adding policies with pod IP pointing to localhost
+			return types.KnoxNetworkLog{}, errors.New("invalid log")
 		}
 
-		if locKnoxLog.Protocol == 0 && locKnoxLog.DstPort == 0 && len(locKnoxLog.DstReservedLabels) == 0 {
-			continue
+		if existingClustername != kaNwLog.ClusterName {
+			_, services, _, pods, err = cluster.GetAllClusterResources(kaNwLog.ClusterName)
+			if err == nil {
+				existingClustername = kaNwLog.ClusterName
+			}
 		}
+		destPod, destLabels, destNs := cluster.ExtractPodSvcInfoFromIP(ip, kaNwLog.ClusterName, pods, services)
+
+		if ip != destPod && (strings.Contains(destPod, "pod") || strings.Contains(destPod, "svc")) {
+			knoxNetLog.DstPodName = strings.Split(destPod, "/")[1]
+			knoxNetLog.DstReservedLabels = strings.Split(destLabels, ",")
+			knoxNetLog.DstNamespace = destNs
+		}
+		knoxNetLog.DstIP = ip
+		knoxNetLog.DstPort, _ = strconv.Atoi(port)
+		knoxNetLog.SynFlag = true
+	} else if strings.Contains(kaNwLog.Data, "SYS_BIND") {
+		var port string
+		// TODO : Identify a way to get protocol from kubearmor
+		// knoxNetLog.Protocol = libs.IPProtocolUDP
+
+		resSlice := strings.Split(kaNwLog.Resource, " ")
```

Comment:
can you please change the variable name, to something else. it can be easily confused with variable `resslice` in line614